### PR TITLE
[#804] Dark Mode

### PR DESCRIPTION
**Pull request by Pukimaa**
Fixing the issue: #804 
Tho maybe also add a button to change it manually but I wanna keep it simple here

### CODE REVIEW

**Comment by waldyrious** [comment]
@Pukimaa can you set up your fork to serve this branch as the source for Github Pages? That way this could be previewed in https://pukimaa.github.io/choosealicense.com.

**Comment by waldyrious** [comment]
For the record, here's how that page looks to me right now:

Screenshot 2021-06-13 at 09 53 10

(Note the active dark mode toggle in Firefox' dev tools, which I tried just in case.)

**Comment by Pukimaa** [comment]
Ok let me push it to GH-Pages...

**Comment by Pukimaa** [comment]
Where does it import the css? I may need to change it because it links to my root of the website and not to the subfolder.

**Comment by waldyrious** [comment]
The stylesheet that the page is trying to load is [https://pukimaa.github.io/assets/css/application.css?v=...](https://pukimaa.github.io/assets/css/application.css?v=7d690890338aac8d654e31e1321608312c593193), whereas the correct link would be [https://pukimaa.github.io/choosealicense.com/assets/css/application.css?v=...](https://pukimaa.github.io/choosealicense.com/assets/css/application.css?v=7d690890338aac8d654e31e1321608312c593193) .

@mlinksva I have very little experience with tweaking Jekyll, but I recall reading in their [GitHub Pages guidelines](https://jekyllrb.com/docs/github-pages/) that it is recommend to use `relative_url` filters for assets. I believe it might make sense to change https://github.com/github/choosealicense.com/blob/b2e3f43e7f051b2d7554e52872d88f2c0231a263/_includes/header.html#L8

to say something like:

```liquid
    <link rel="stylesheet" href="{{ /assets/css/application.css | relative_url }}?v={{ site.github.build_revision }}"> 
```

**Comment by waldyrious** [comment]
In fact, it's also necessary to add the `relative_url` filter to https://github.com/github/choosealicense.com/blob/b2e3f43e7f051b2d7554e52872d88f2c0231a263/_includes/header.html#L9

and probably the import URLs at https://github.com/github/choosealicense.com/blob/b2e3f43e7f051b2d7554e52872d88f2c0231a263/assets/css/application.scss#L4-L5 

would need to become:

```css
@import "../vendor/normalize-css/normalize.css";
@import "../vendor/qtip2/jquery.qtip.min.css";
```

**Comment by Pukimaa** [comment]
Did all you said...lets wait for it to be build...

**Comment by Pukimaa** [comment]
didnt worked...

**Comment by waldyrious** [comment]
As I said, I'm no Jekyll expert. Maye  try omitting the leading `/` in the path when using the `relative_url` filter? In any case, I'm sure @mlinksva will be able to help.

**Comment by mlinksva** [comment]
@Pukimaa I suspect https://github.com/github/choosealicense.com/compare/gh-pages...Pukimaa:gh-pages#diff-c0311c1df17d5d714db46ca1401ad374a35de861a461d211c6c947f1b2194034R8 is not right.

Look at @waldyrious suggestion at https://github.com/github/choosealicense.com/pull/833#issuecomment-860777503 again.

**Comment by waldyrious** [comment]
Fixed link of the diff ([thanks to Refined GitHub](https://github.com/sindresorhus/refined-github/pull/3085)): [`gh-pages...Pukimaa:gh-pages`#diff-c0311c1df1](https://github.com/github/choosealicense.com/compare/gh-pages...Pukimaa:gh-pages#diff-c0311c1df17d5d714db46ca1401ad374a35de861a461d211c6c947f1b2194034R8).

**Comment by Pukimaa** [comment]
sry it did worked just GH Pages was a little bit slow. Also I just saw I need to add some importants...if I am finished I make another commit and the work is done ^^

**Comment by Pukimaa** [comment]
ok made a new commit for `gh-pages` for me to review if it looks good I will also push it to the `patch-1` branch

**Comment by mlinksva** [comment]
https://pukimaa.github.io/choosealicense.com/ is still not loading CSS. I suspect it's still due to https://github.com/github/choosealicense.com/compare/gh-pages...Pukimaa:gh-pages#diff-c0311c1df17d5d714db46ca1401ad374a35de861a461d211c6c947f1b2194034R8 ... again, compare with @waldyrious suggestion above, and actually, the docs (I think the suggestion has a leading `/` that shouldn't be there).

I think the line referenced above should be

```
    <link rel="stylesheet" href="{{ assets/css/application.css | relative_url }}?v={{ site.github.build_revision }}">
```

**Review by TejasAvinashShetty — APPROVED** [review]
nice. Seems ok

**Comment by Pukimaa** [comment]
ehhhm the text is not updated yet I think and some other stuff needs some adjustments

**Comment by mlinksva** [comment]
@Pukimaa what text updates or other adjustments are you referring to? Now that the site renders correctly in a subdirectory, dark mode looks pretty good to me at https://pukimaa.github.io/choosealicense.com and https://pukimaa.github.io/choosealicense.com/licenses/mit/

Screen Shot 2021-08-13 at 12 23 30 PM

Well done!

Let me know what you think still needs to be done to ship this. Anyone else with reviews or other thoughts, most welcome.

**Comment by waldyrious** [comment]
The `{` and `}` in the home page headings are much more conspicuous in the dark mode than in the light mode:

| Light | Dark |
| ----- | ---- |
| Screenshot from 2021-08-13 20-38-39 | Screenshot from 2021-08-13 20-38-53 |

IMO changing https://github.com/github/choosealicense.com/blob/3eac53eff0af04b9598f07ff5dd12b512a20729f/assets/css/application.scss#L95-L96 like this, for example:

```diff
-  color: #f1f1f1;
+  color: rgba(127, 127, 127, 10%);
```

...would make the effect more consistent:

| Light | Dark |
| ----- | ---- |
| Screenshot from 2021-08-13 20-45-26 | Screenshot from 2021-08-13 20-45-36 |

File: assets/css/application.scss [hunk 1]
```diff
@@ -10,6 +10,46 @@ body {
...
+    color: #ddd !important; 
+  }
+  
+  .home h2 {
+    color: #c7cdce !important;
+  }
+  
+  .license-body pre {
+	background-color: #131313 !important;
+	border: 1px solid #3e3e3e !important;
```
Comment (waldyrious) [inline assets/css/application.scss]: Just a nitpick, but these two lines are indented with tabs while the others are indented with spaces. (The same happens in line 41, 45 and 49.)

**Review by WordLife617 — APPROVED** [review]
seems good...Great job! keep up the good work

**Comment by mlinksva** [comment]
Thanks @Pukimaa and @waldyrious! I'm going to merge this now, happy to consider improvements in followup PRs as needed.

**Comment by Pukimaa** [comment]
Yay I'm going to update the { too with a new pr today

**Pull request merged**